src/Lumi/Components2/QRCode.purs

```diff
@@ -0,0 +1,67 @@
+module Lumi.Components2.QRCode where
+
+import Prelude
+import Data.Newtype (class Newtype)
+import Data.Nullable as Nullable
+import Effect (Effect)
+import Lumi.Styles (toCSS)
+import Lumi.Styles.QRCode as Styles.QRCode
+import Lumi.Styles.Theme (LumiTheme, useTheme)
+import React.Basic.DOM as R
+import React.Basic.Emotion as E
+import React.Basic.Hooks (Hook, JSX, ReactComponent, Ref, UseContext, UseRef, coerceHook, element, useRef)
+import React.Basic.Hooks as React
+import Web.DOM (Node)
+
+newtype UseQRCode hooks
+  = UseQRCode (UseRef (Nullable.Nullable Node) (UseContext LumiTheme hooks))
+
+derive instance ntUseQRCode :: Newtype (UseQRCode hooks) _
+
+data ErrorCorrectLevel
+  = L
+  | M
+  | Q
+  | H
+
+errorCorrectLevelToString :: ErrorCorrectLevel -> String
+errorCorrectLevelToString = case _ of
+  L -> "L"
+  M -> "M"
+  Q -> "Q"
+  H -> "H"
+
+useQRCode :: String -> ErrorCorrectLevel -> Hook UseQRCode { qrcode :: JSX, saveOnClick :: String -> Effect Unit }
```

Comment:
Can we make the hook return `qrcode` as a `LumiComponent`? This would allow for consumers to apply custom styling with `PropModifier`s.

Reply:
We could. This seemed a little simpler and if it's just for layout/sizing purposes the parent can wrap it with a box and style that. But you're right, that would be a bit more powerful.

Reply:
I think that if it's a low effort change we should definitely consider it for a first release of this component, so as to avoid future breaking changes.